presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkStaticQueryExecution.java

```diff
@@ -0,0 +1,347 @@
+                        DataSize.succinctBytes(currentFragmentOutputUncompressedSizeInBytes));
+                totalNumberOfPagesReceived += pages.size();
+                totalCompressedSizeInBytes += currentFragmentOutputCompressedSizeInBytes;
+                totalUncompressedSizeInBytes += currentFragmentOutputUncompressedSizeInBytes;
+                inputs.put(inputFuture.getKey(), pages);
+            }
+
+            log.info(
+                    "Received %s pages in total. Compressed size: %s. Uncompressed size: %s.",
+                    totalNumberOfPagesReceived,
+                    DataSize.succinctBytes(totalCompressedSizeInBytes),
+                    DataSize.succinctBytes(totalUncompressedSizeInBytes));
+
+            IPrestoSparkTaskExecutor<PrestoSparkSerializedPage> prestoSparkTaskExecutor = taskExecutorFactory.create(
+                    0,
+                    0,
+                    serializedTaskDescriptor,
+                    emptyScalaIterator(),
+                    new PrestoSparkTaskInputs(ImmutableMap.of(), ImmutableMap.of(), inputs.build()),
+                    taskInfoCollector,
+                    shuffleStatsCollector,
+                    PrestoSparkSerializedPage.class);
+            return collectScalaIterator(prestoSparkTaskExecutor);
+        }
+
+        RddAndMore<PrestoSparkSerializedPage> rootRdd = createRdd(rootFragmentedPlan, PrestoSparkSerializedPage.class, tableWriteInfo);
+        return rootRdd.collectAndDestroyDependenciesWithTimeout(computeNextTimeout(queryCompletionDeadline), MILLISECONDS, waitTimeMetrics);
+    }
+
+    private SubPlan getFragmentedPlan()
+    {
+        SubPlan rootFragmentedPlan = planFragmenter.fragmentQueryPlan(session, planAndMore.getPlan(), warningCollector);
+        queryMonitor.queryUpdatedEvent(
+                createQueryInfo(
+                        session,
+                        query,
+                        PLANNING,
+                        Optional.of(planAndMore),
+                        sparkQueueName,
+                        Optional.empty(),
+                        queryStateTimer,
+                        Optional.of(createStageInfo(session.getQueryId(), rootFragmentedPlan, ImmutableList.of())),
+                        warningCollector));
+
+        log.info(textDistributedPlan(rootFragmentedPlan, metadata.getFunctionAndTypeManager(), session, true));
+        int hashPartitionCount = getHashPartitionCount(sparkContext.sc(), session.getQueryId(), session, planAndMore);
+        rootFragmentedPlan = configureOutputPartitioning(session, rootFragmentedPlan, hashPartitionCount);
+        return rootFragmentedPlan;
+    }
+
+    @Override
+    protected SubPlan getCurrentFragmentedPlan()
+    {
+        return currentFragmentedPlan;
+    }
+
+    private int getHashPartitionCount(SparkContext sparkContext, QueryId queryId, Session session, PlanAndMore planAndMore)
```

Comment:
These methods might be reused across implementations of the PrestoSparkQueryExecutor but we can factor them out when we get there.